lib/src/insertion_order_parser.dart

```diff
@@ -0,0 +1,199 @@
+      period.name: period
+  };
+
+  static final _pacingTypeMap = {
+    for (var type in InsertionOrder_Pacing_PacingType.values) type.name: type
+  };
+
+  static final _budgetUnitMap = {
+    for (var unit in InsertionOrder_InsertionOrderBudget_BudgetUnit.values)
+      unit.name: unit
+  };
+
+  static final _automationTypeMap = {
+    for (var type
+        in InsertionOrder_InsertionOrderBudget_InsertionOrderAutomationType
+            .values)
+      type.name: type
+  };
+
+  /// Parses a json string to [InsertionOrder].
+  static InsertionOrder parse(String jsonString) {
+    Map<String, dynamic> map = json.decode(jsonString);
+    return _createInsertionOrder(map);
+  }
+
+  /// Creates an [InsertionOrder] instance from [map].
+  static InsertionOrder _createInsertionOrder(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) return InsertionOrder();
+
+    final insertionOrder = InsertionOrder()
+      ..advertiserId = map['advertiserId'] ?? _emptyEntry
+      ..campaignId = map['campaignId'] ?? _emptyEntry
+      ..insertionOrderId = map['insertionOrderId'] ?? _emptyEntry
+      ..displayName = map['displayName'] ?? _emptyEntry
+      ..updateTime = map['updateTime'] ?? _emptyEntry
+      ..entityStatus = _createEntityStatus(map['entityStatus'])
+      ..pacing = _createPacing(map['pacing'])
+      ..budget = _createBudget(map['budget']);
+
+    return insertionOrder;
+  }
+
+  /// Creates an [InsertionOrder_Pacing] instance from [map].
+  static InsertionOrder_Pacing _createPacing(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) return InsertionOrder_Pacing();
+
+    final pacing = InsertionOrder_Pacing()
+      ..pacingPeriod = _createPacingPeriod(map['pacingPeriod'])
+      ..pacingType = _createPacingType(map['pacingType'])
+      ..dailyMaxMicros = map['dailyMaxMicros'] ?? _emptyEntry
+      ..dailyMaxImpressions = map['dailyMaxImpressions'] ?? _emptyEntry;
+
+    return pacing;
+  }
+
+  /// Creates an [InsertionOrder_InsertionOrderBudget] instance from [map].
+  static InsertionOrder_InsertionOrderBudget _createBudget(
+      Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget();
+    }
+
+    final budget = InsertionOrder_InsertionOrderBudget()
+      ..budgetUnit = _createBudgetUnit(map['budgetUnit'])
+      ..automationType = _createAutomationType(map['automationType']);
+
+    for (Map segmentMap in map['budgetSegments'] ?? []) {
+      budget.budgetSegments.add(_createBudgetSegment(segmentMap));
+    }
+    return budget;
+  }
+
+  /// Creates an
+  /// [InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment]
+  /// instance from [map].
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment
+      _createBudgetSegment(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment();
```

Comment:
did you want to return this?
did a test go through this branch?

Reply:
ugh now this is tested